codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/auth/SigV4AsymmetricAuthSchemeIntegration.kt

```diff
@@ -34,6 +39,40 @@ class SigV4AsymmetricAuthSchemeIntegration : KotlinIntegration {
 
     override fun authSchemes(ctx: ProtocolGenerator.GenerationContext): List<AuthSchemeHandler> =
         if (modelHasSigV4aTrait(ctx)) listOf(SigV4AsymmetricAuthSchemeHandler()) else emptyList()
+
+    override fun additionalServiceConfigProps(ctx: CodegenContext): List<ConfigProperty> =
+        if (modelHasSigV4aTrait(ctx)) {
```

Comment:
**question**: Why can't this be determined in `enabledForService`? I see a comment about how integrations are filtered but I'm still not sure I'm following.

Reply:
relevant: https://github.com/awslabs/aws-sdk-kotlin/pull/1245#discussion_r1515002007

Reply:
SigV4a as a trait is added for some services that don't model their use in another integration, `SigV4AsymmetricTraitCustomization`. We could look for that trait in `enabledForService` but `CodegenVisitor` filters integrations out based on that `enabledForService` function before the model preprocessing can happen from integration `SigV4AsymmetricTraitCustomization`. The fix is simple but it breaks some seemingly unrelated tests.

Reply:
`CodegenVisitor.kt`
```
integrations = ServiceLoader.load(KotlinIntegration::class.java, classLoader)
    .onEach { integration -> LOGGER.info("Loaded KotlinIntegration: ${integration.javaClass.name}") }
    .filter { integration -> integration.enabledForService(context.model, settings) } // TODO: Change so we don't filter until previous integrations model modifications are complete
    .onEach { integration -> LOGGER.info("Enabled KotlinIntegration: ${integration.javaClass.name}") }
    .sortedBy(KotlinIntegration::order)


LOGGER.info("Preprocessing model")
// Model pre-processing:
// 1. Start with the model from the plugin context
// 2. Apply integrations
// 3. Flatten error shapes (see: https://github.com/awslabs/smithy/pull/919)
// 4. Normalize the operations
var resolvedModel = context.model
for (integration in integrations) {
    resolvedModel = integration.preprocessModel(resolvedModel, settings)
}

```

Reply:
I'm working on those tests right now btw

Reply:
This is fixed in [smithy-kotlin#1049](https://github.com/awslabs/smithy-kotlin/pull/1049)